packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,241 @@
+    protected animation!: Vue;
+
+    /**
+     * Time in milliseconds during which each message is visible.
+     *
+     * @public
+     */
+    @Prop({ default: 1500 })
+    protected animationIntervalMs!: number;
+
+    /**
+     * Whether the messages animation is active only when hovering the search input or always.
+     *
+     * @public
+     */
+    @Prop({ default: false })
+    protected animateOnlyOnHover!: boolean;
+
+    @State('searchBox', 'query')
+    public query!: string;
+
+    protected isSearchBoxHovered = false;
+
+    protected isSearchBoxFocused = false;
+
+    protected animationMessageIndex = 0;
+
+    protected animationInterval: number | undefined;
+
+    protected get isVisible(): boolean {
+      return !this.query && !this.isSearchBoxFocused;
+    }
+
+    protected get isBeingAnimated(): boolean {
+      return this.isVisible && (!this.animateOnlyOnHover || this.isSearchBoxHovered);
+    }
+
+    protected get animationMessage(): string | undefined {
+      return this.messages[this.animationMessageIndex];
+    }
+
+    protected resetAnimation(): void {
+      if (this.isBeingAnimated) {
+        this.restartAnimationInterval();
+      } else {
+        this.stopAnimationInterval();
+      }
+    }
+
+    protected restartAnimationInterval(): void {
+      this.stopAnimationInterval();
+
+      this.animationInterval = setInterval((): void => {
+        this.incrementAnimationMessageIndex();
+      }, this.animationIntervalMs);
+    }
+
+    protected stopAnimationInterval(): void {
+      if (!this.animationInterval) {
+        return;
+      }
+      clearInterval(this.animationInterval);
+      this.animationInterval = undefined;
+    }
+
+    protected incrementAnimationMessageIndex(): void {
+      this.animationMessageIndex = (this.animationMessageIndex + 1) % this.messages.length;
+    }
+
+    @XOn('UserHoveredInSearchBox')
+    onUserHoveredInSearchBox(): void {
+      this.isSearchBoxHovered = true;
+    }
+
+    @XOn('UserHoveredOutSearchBox')
+    onUserHoveredOutSearchBox(): void {
+      this.isSearchBoxHovered = false;
+    }
+
+    @XOn('UserFocusedSearchBox')
+    onUserFocusedSearchBox(): void {
+      this.isSearchBoxFocused = true;
+    }
+
+    @XOn('UserBlurredSearchBox')
+    onUserBlurredSearchBox(): void {
+      this.isSearchBoxFocused = false;
+    }
+
+    mounted(): void {
+      this.resetAnimation();
+
+      this.$watch('isBeingAnimated', (value: boolean) => {
+        if (!value) {
+          this.incrementAnimationMessageIndex();
+        }
+        this.resetAnimation();
+      });
+
+      this.$watch('messages', () => {
+        this.animationMessageIndex = 0;
+        this.resetAnimation();
+      });
+
+      this.$watch('animationIntervalMs', () => {
+        this.resetAnimation();
+      });
+    }
+
+    beforeDestroy(): void {
+      this.stopAnimationInterval();
+    }
+  }
+</script>
+
+<style lang="scss" scoped>
+  .x-search-input-placeholder {
```

Comment:
I am not sure about adding these styles here, maybe I will set this values in the design system or inline like:

```
    <SearchInputPlaceholder
                  :messages="searchInputPlaceholderMessages"
                  :animate-only-on-hover="true"
                  class="x-list x-list--justify-center"
                >
 ```